include/wlr/types/wlr_surface.h

```diff
@@ -109,6 +109,9 @@ struct wlr_surface {
 	// wlr_subsurface::parent_pending_link
 	struct wl_list subsurface_pending_list;
 
+	// wlr_keyboard_shortcuts_inhibitor_v1
+	struct wl_list keyboard_shortcuts_inhibitors;
```

Comment:
We try not to leak protocol implementation details in `wlr_surface`. Moreover, the protocol is unstable, so new versions will change how this works.

Reply:
I expected as much but it was the one way I could think of to avoid bookkeeping by and passing around of the manager and inhibitors within the compositor. Can you recommend another way of  implementing similar properties? Or is this the way to go: Hand the manager and inhibitors to the compositor and tell it to get that information to where it needs it its own way?

Reply:
The latter might work. It could also be possible to store the surface on the inhibitor and provide a convenience function which answers "does this surface have an inhibitor?" by simply iterating through all of the inhibitors.

Reply:
This is what I ended up doing in the updated sway implementation: I've effectively moved the per-surface backlink and convenience function into the compositor. I've put that up for discussion as swaywm/sway#5021.